python_modules/dagster/dagster/serdes/serdes.py

```diff
@@ -115,18 +119,28 @@ def whitelist_for_serdes(serializer):
 
     """
 
-    if isclass(serializer) and not issubclass(serializer, Serializer):
-        return _whitelist_for_serdes(whitelist_map=_WHITELIST_MAP, serializer=None)(serializer)
+    if __cls is not None:
+        check.class_param(__cls, "__cls")
+        return _whitelist_for_serdes(whitelist_map=_WHITELIST_MAP, serializer=None)(__cls)
     else:
         check.subclass_param(serializer, "serializer", Serializer)
         serializer = cast(Type[Serializer], serializer)
         return _whitelist_for_serdes(whitelist_map=_WHITELIST_MAP, serializer=serializer)
 
+    # # target (no args passed to decorator)
+    # if isclass(serializer) and not issubclass(serializer, Serializer):
+    #     return _whitelist_for_serdes(whitelist_map=_WHITELIST_MAP, serializer=None)(serializer)
+    # # serializer (args passed to decorator)
+    # else:
+    #     check.subclass_param(serializer, "serializer", Serializer)
+    #     serializer = cast(Type[Serializer], serializer)
+    #     return _whitelist_for_serdes(whitelist_map=_WHITELIST_MAP, serializer=serializer)
+
```

Comment:
whats going on here?

Reply:
Whoops, that was the old logic, can be removed.